Wikipedia/Code/WMFBarButtonItemPopoverBackgroundView.m

```diff
@@ -1,121 +1,15 @@
 #import "WMFBarButtonItemPopoverBackgroundView.h"
+#import "WMFPopoverBackgroundView.h"
 #import "UIColor+WMFStyle.h"
 
-@interface WMFBarButtonItemPopoverBackgroundView ()
-
-@property (nonatomic, readwrite) CGFloat arrowOffset;
-@property (nonatomic, readwrite) UIPopoverArrowDirection arrowDirection;
-@property (nonatomic, strong) UIImageView *arrowImageView;
-
-@end
-
 @implementation WMFBarButtonItemPopoverBackgroundView
 
-@synthesize arrowDirection = _arrowDirection;
-@synthesize arrowOffset = _arrowOffset;
-
 - (instancetype)initWithFrame:(CGRect)frame {
     self = [super initWithFrame:frame];
     if (self) {
-        self.arrowDirection = UIPopoverArrowDirectionAny;
-        self.arrowImageView = [[UIImageView alloc] initWithImage:[self arrowImageWithSize:CGSizeMake([[self class] arrowBase], [[self class] arrowHeight])]];
-        [self addSubview:self.arrowImageView];
+        self.arrowColor = [UIColor wmf_barButtonItemPopoverMessageBackgroundColor];
```

Comment:
Sorry my last comment may be a little unclear, just a TL;DR
I think the superclass is good. - I think this subclass is unneeded since it is just one line.